### PR TITLE
Use ${Python3_EXECUTABLE} instead of shebang on top of Python files

### LINKED ISSUE #3680: Invoke python all via Python3_EXECUTABLE
**Describe the bug**
```
../CMake/compareGCOV.cmake:  execute_process(COMMAND python ${CTEST_SOURCE_DIRECTORY}/tests/coverage/compare_gcov.py --action merge --base-dir
../CMake/compareGCOV.cmake:  execute_process(COMMAND python ${CTEST_SOURCE_DIRECTORY}/tests/coverage/compare_gcov.py --action compare --base-dir
../CMake/run_pyscf.cmake:  add_pyscf_test(${BASE_NAME} python ${MY_WORKDIR} ${TEST_INPUT_PREFIX}.py)
```

```
../CMake/macros.cmake:            ${qmcpack_SOURCE_DIR}/tests/scripts/check_scalars.py
../CMake/test_labels.cmake:      COMMAND ${qmcpack_SOURCE_DIR}/tests/scripts/test_labels.py ${TEST_NAME} ${QMC_CUDA} ${QMC_COMPLEX}
```

There are more if I looked into tests folder.

Are there ways to stop running python via shebang so we can caught all places?

### PR DESCRIPTION
**Pull request by quantumsteve**
Please review the [developer documentation](https://github.com/QMCPACK/qmcpack/wiki/Development-workflow)
on the wiki of this project that contains help and requirements.

## Proposed changes

Describe what this PR changes and why.  If it closes an issue, link to it here
with [a supported keyword](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

This removes the shebang on top of Python files. CMake was updated to use ${Python3_EXECUTABLE}.

I avoided files in the nexus sub-directory.

Fixes #3680

## What type(s) of changes does this code introduce?
_Delete the items that do not apply_

- Refactoring (no functional changes, no api changes)
- Build related changes

### Does this introduce a breaking change?

- No

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

**Comment by prckent** [comment]
There are a lot of utilities and examples touched here that are not run by CMake. Many of the same considerations as for Nexus apply. However there is an argument for requiring the "correct" python to have to be specified by the user.

File: CMake/macros.cmake [hunk 1]
```diff
@@ -400,7 +400,7 @@ else(QMC_NO_SLOW_CUSTOM_TESTING_COMMANDS)
               #MESSAGE("check command = ${CHECK_CMD}")
               add_test(
                 NAME ${TEST_NAME}
-                COMMAND ${CHECK_CMD}
+		COMMAND ${Python3_EXECUTABLE} ${CHECK_CMD}
```
Comment (ye-luo) [inline CMake/macros.cmake:403]: tabs.

**Review by ye-luo — COMMENTED** [review]
examples, labs and utils need to keep the shebang as they are not invoked by CMake/CTest.

**Comment by markdewing** [comment]
Keeping all the shebangs would be a reasonable (and safe) option.  Removing them all seems like a temporary hack to find all places python scripts are called from cmake to address #3680.

**Comment by jtkrogel** [comment]
I think keeping she-bangs in anything a user might invoke is desirable.  If the default provided by the she-bang is somehow wrong for their environment, they can always invoke `the_correct_python_exe script`.  This is just about sensible defaults and convenience.

**Review by ye-luo — APPROVED** [review]

**Review by prckent — APPROVED** [review]
List of modified files LGTM

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**